modules/cache-common/src/main/java/org/opensearch/cache/common/policy/TookTimePolicy.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.cache.common.policy;
+
+import org.opensearch.common.unit.TimeValue;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * A cache tier policy which accepts queries whose took time is greater than some threshold.
+ * The threshold should be set to approximately
+ * the time it takes to get a result from the cache tier.
+ * The policy accepts values of type V and decodes them into CachePolicyInfoWrapper, which has the data needed
```

Comment:
We should remove the usage of `CachePolicyInfoWrapper` from here and other places as that is not used anymore. Also doc should be updated to reflect new parser changes.

Reply:
Oops - I'd missed this one, good catch.